rskj-core/src/test/java/co/rsk/peg/bitcoin/BitcoinUtilsTest.java

```diff
@@ -181,14 +181,65 @@ void extractRedeemScriptFromInput_forP2shP2wshTx_withInvalidRedeemScript_shouldR
+        Script redeemScript = new Script(redeemScriptHex);
+
+        // Act
+        Optional<Sha256Hash> sigHash = getFirstInputSigHash(btcTx);
+
+        // Assert
+        assertTrue(sigHash.isPresent());
+        Sha256Hash expectedSigHash = btcTx.hashForSignature(
+            0,
+            redeemScript,
+            BtcTransaction.SigHash.ALL,
+            false
+        );
+        assertEquals(expectedSigHash, sigHash.get());
+    }
+
+    @Test
+    void getFirstInputSigHash_no_input() {
+        // Arrange
+        BtcTransaction btcTx = new BtcTransaction(btcMainnetParams);
+        btcTx.addOutput(Coin.COIN, destinationAddress);
+
+        // Act
+        Optional<Sha256Hash> sigHash = BitcoinUtils.getFirstInputSigHash(btcTx);
+
+        // Assert
+        assertFalse(sigHash.isPresent());
+    }
+
+    @Test
+    void getFirstInputSigHash_invalid_input_no_redeem_script() {
+        // Arrange
+        BtcTransaction btcTx = new BtcTransaction(btcMainnetParams);
+        btcTx.addInput(BitcoinTestUtils.createHash(1), 0, new Script(new byte[]{}));
+        btcTx.addOutput(Coin.COIN, destinationAddress);
+
+        // Act
+        Optional<Sha256Hash> sigHash = BitcoinUtils.getFirstInputSigHash(btcTx);
+
+        // Assert
+        assertFalse(sigHash.isPresent());
```

Comment:
```suggestion
     assertTrue(sigHash.isEmpty());
```

Wdyt? We tend to prefer positive conditions over negative ones.